rust/kernel/fs.rs

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! A kernel file system.
+//!
+//! This module allows Rust code to register new kernel file systems.
+//!
+//! C headers: [`include/linux/fs.h`](../../include/linux/fs.h)
+
+/// Read-only file systems.
+pub mod ro {
+    use crate::error::{code::*, from_result, to_result, Error};
+    use crate::types::Opaque;
+    use crate::{bindings, init::PinInit, str::CStr, try_pin_init, ThisModule};
+    use core::{marker::PhantomPinned, pin::Pin};
+    use macros::{pin_data, pinned_drop};
+
+    /// A read-only file system type.
+    pub trait Type {
```

Comment:
Since this is only referenced once in modules, I expect this to be referenced like:

```rust
impl fs::ro::Type for T {
}
```

Or in the future when we make it rw:

```rust
impl fs::Type for T {
}
```
So it would be clear that it's a file system type. In fact, the name in C is `file_system_type`.

But I'm also ok with `FileSystem`.

Reply:
At the moment you also often just use `Type` and users are not forced to use `fs::ro::cType`, so I would prefer a more meaningful name.